packages/flutter/lib/src/material/button_style_button.dart

```diff
@@ -191,33 +194,49 @@ abstract class ButtonStyleButton extends StatefulWidget {
 ///  * [TextButton], a simple button without a shadow.
 ///  * [ElevatedButton], a filled button whose material elevates when pressed.
 ///  * [OutlinedButton], similar to [TextButton], but with an outline.
-class _ButtonStyleState extends State<ButtonStyleButton> with MaterialStateMixin, TickerProviderStateMixin {
-  AnimationController? _controller;
-  double? _elevation;
-  Color? _backgroundColor;
+class _ButtonStyleState extends State<ButtonStyleButton> with TickerProviderStateMixin {
+  AnimationController? controller;
+  double? elevation;
+  Color? backgroundColor;
+  late final MaterialStatesController statesController;
+
+  void handleStatesControllerChange() {
+    // Force a rebuild to resolve MaterialStateProperty properties
+    setState(() { });
```

Comment:
I assume that it's up to the caller to make sure that they don't call `update` on a `MaterialStateController` unless they are currently allowed to rebuild this widget (either inside this widget or a parent currently building)?

Also, maybe checking to see if this widget is still mounted might be helpful? Although I guess we're unregistered in dispose, so it probably can't happen.

Reply:
Yes, it must be safe to call setState() when the ButtonStyleButton's controller's states are changed.

If setState() is called at the wrong time the error message is pretty self-explanatory